Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <optional>
+
+#include <QCheckBox>
+#include <QDialog>
+#include <QGroupBox>
+#include <QLineEdit>
+#include <QWidget>
+
+#include "Core/Core.h"
+#include "Core/IOS/USB/Emulated/Skylander.h"
+
+class QDialogButtonBox;
+class QLabel;
+class QPushButton;
+class QSpinBox;
+class QTabWidget;
+
+class SkylanderPortalWindow : public QWidget
+{
+  Q_OBJECT
+public:
+  explicit SkylanderPortalWindow(QWidget* parent = nullptr);
+  ~SkylanderPortalWindow();
+
+protected:
+  std::array<QLineEdit*, MAX_SKYLANDERS> edit_skylanders;
+  std::array<std::optional<std::tuple<u8, u16, u16>>, MAX_SKYLANDERS> sky_slots;
```

Comment:
`#include <utility>` (for tuple)

Reply:
Actually, consider making a struct so that each field can be named. It's unclear what u8, u16, u16 each represent for someone who hasn't looked at the .cpp file.